crates/engine/tree/src/tree/mod.rs

```diff
@@ -342,6 +368,62 @@ where
                     }
                 }
             }
+
+            if self.should_persist() {
+                self.persist_state().await;
```

Comment:
this is likely problematic, because now we're waiting until this action is done, preventing us from handling new requests.

I think it's reasonable to accept that only one persistence action will be action at any time?
so I think we can instead store the receiver and check via [try_recv](https://docs.rs/tokio/latest/tokio/sync/mpsc/struct.Receiver.html#method.try_recv) ? 
this would mean we need to do a bit more work on tracking active persistence actions

Reply:
indeed, that's a very good point, done here https://github.com/paradigmxyz/reth/pull/9365/commits/875de2f2d52e8f154a7f362ff15b3054cc7e5656 ptal